### ISSUE TITLE
[Python Suggestion] Linting

### ISSUE THREAD

**Issue by adehad** [edgedb/edgedb-python#438]
Hi there!

I was wondering if there was any interest in general python project related improvements/contributions?

I made a linting proof of concept with [pre-commit](pre-commit.ci), a great tool I use in all my project (even non-Python). I have added a `lint` target in the Makefile to reflect this intention.

https://github.com/adehad/edgedb-python/pull/1/files

There are quite a few things it runs, but I think the autoformatters `black` and `ruff` can be a great asset to run (can even be configured via the pre-commit github action that can autoformat pull requests). `ruff` can be used to swap out `flake8` to run on the generated file too if desired.
Ideally the `mypy` (static type checker) should be run, but looks like there are quite a few errors at the moment.

Happy to help (maybe in smaller increments) if this is something you are considering and/or need help with

**Comment by fantix** [comment]
Yeah, I think that'll be great! Thank you for the proposal, feel free to go ahead with incremental PRs. :heart:

> Ideally the mypy (static type checker) should be run, but looks like there are quite a few errors at the moment.

Agreed! Let's fix the errors and turn it on.
